src/CarbonAware/test/Configuration/DataSourcesConfigurationTests.cs

```diff
@@ -57,37 +57,40 @@ public void AssertValid_DataSourceNotInConfigurationSection_ThrowsException(stri
         Assert.That(ex!.Message, Contains.Substring(errorMessage));
     }
 
-    [Test]
-    public void ConfigurationTypesAndSections_ReturnCorrectType()
+    [TestCase("WattTime", "Json", TestName = "Assert Configuration Type is forecast=WattTime and emissions=Json")]
+    [TestCase("WattTime", "ElectricityMaps", TestName = "Assert Configuration Type is forecast=WattTime and emissions=ElectricityMaps")]
+    [TestCase("Json", null, TestName = "Assert Configuration Type is forecast=Json and emissions=null")]
+    public void ConfigurationTypesAndSections_ReturnCorrectType(string forecastDataSource, string emissionsDataSource)
     {
         // Arrange
         var inMemorySettings = new Dictionary<string, string>()
         {
             { "Configurations:WattTime:Type", "WattTime" },
-            { "Configurations:Json:Type", "Json" }
+            { "Configurations:Json:Type", "Json" },
```

Comment:
Should be these configuration params also use what is passed through the parameters of the method? `forecastDataSource` and `emissionsDataSource`?
```c#
public void ConfigurationTypesAndSections_ReturnCorrectType(string confType)
{
...
{ $"Configuration:{confType}:Type", $"{confType}"},
...
}
```

Reply:
We could do that, but we don't necessarily need to. I decided to include all three types and remove the need to handle the case where a confType for one of the data sources if null from the testcase. Let me know if you see a better case for doing it this way though!